src/js/views/admin/actions/actions_views.js

```diff
@@ -49,33 +54,57 @@ const PublishedComponent = Droplist.extend({
     const isCompact = this.getOption('isCompact');
 
     return {
-      className: isCompact ? 'button-secondary--compact' : 'button-secondary w-100',
-      template: PublishedTemplate,
+      tagName: 'button',
+      className: isCompact ? 'button-secondary--compact is-icon-only' : 'button-secondary w-100',
+      template: this.getTemplate(),
     };
   },
   picklistOptions: {
     headingText: i18n.publishedComponent.headingText,
     getItemFormat({ attributes }) {
-      return new Handlebars.SafeString(PublishedTemplate(attributes));
+      const template = hbs`<span class="{{ className }}">{{far icon}}{{ name }}</span>`;
+      return new Handlebars.SafeString(template(attributes));
     },
   },
 });
 
 const OwnerItemTemplate = hbs`<a{{#if isSelected}} class="is-selected"{{/if}}>{{matchText name query}} <span class="program-actions__role">{{matchText short query}}</span></a>`;
 
 const OwnerComponent = Droplist.extend({
+  isCompact: false,
+  getTemplate() {
+    if (this.getOption('isCompact')) {
+      return hbs`{{far "user-circle"}}{{ name }}`;
+    }
+
+    return hbs`{{far "user-circle"}}{{ name }}{{#unless name}}{{ @intl.admin.actions.actionsViews.ownerComponent.defaultText }}{{/unless}}`;
+  },
   popWidth() {
     return this.getView().$el.outerWidth();
   },
   picklistOptions: _.extend({
     isSelectlist: true,
   }, i18n.ownerComponent),
-  viewOptions: {
-    modelEvents: {
-      'change:_role': 'render',
-    },
-    className: 'button-secondary w-100',
-    template: hbs`{{far "user-circle"}}{{ name }}{{#unless name}}{{ @intl.admin.actions.actionsViews.ownerComponent.defaultText }}{{/unless}}`,
+  viewOptions() {
+    const isCompact = this.getOption('isCompact');
+    const name = this.getState('selected');
+    return {
+      modelEvents: {
+        'change:_role': 'render',
+      },
+      className() {
+        if (!name && isCompact) {
```

Comment:
so when I log out `this.model` inside of `className()` it's `undefined` no matter the value of the owner

Reply:
ah.. right because `className` happens prior to the `initialize`.. hmm so it'd have to be `this.getOption('state').get('selected')`  😐 ok well maybe just call it `selected`